models/documentlistmodel.cpp

```diff
@@ -97,15 +98,32 @@ QHash< int, QByteArray > DocumentListModel::roleNames() const
     return d->roles;
 }
 
+void DocumentListModel::setAllItemsDirty(bool status)
+{
+    for(QList<DocumentListModelEntry>::iterator entry = d->entries.begin();
+        entry != d->entries.end(); entry++) {
+        entry->dirty = status;
+    }
+}
+
 void DocumentListModel::addItem(QString name, QString path, QString type, int size, QDateTime lastRead, QString mimeType)
 {
     // We sometimes get duplicate entries... and that's kind of silly.
-    foreach(const DocumentListModelEntry& entry, d->entries) {
-        if(entry.filePath == path)
+    for(QList<DocumentListModelEntry>::iterator entry = d->entries.begin();
+        entry != d->entries.end(); entry++) {
+        if(entry->filePath == path) {
+            entry->dirty = false;
+            entry->fileType = type;
+            entry->fileSize = size;
+            entry->fileRead = lastRead;
+            entry->mimeType = mimeType;
+            entry->documentClass = static_cast<DocumentClass>(mimeTypeToDocumentClass(mimeType));
```

Comment:
Thanks for reviewing the patch.

Here, I don't check all fields, the other are affectation. It is to handle the second point in my patch comment : i.e. when an entry change type or other field. For instance I've already a file /path/to/bill.pdf and I save a file to the same path, but it's a new bill.pdf, thus fields must be updated.